source/h2_connection.c

```diff
@@ -1575,28 +1615,70 @@ int aws_h2_connection_on_stream_closed(
     return AWS_OP_SUCCESS;
 }
 
+static struct aws_h2err s_remove_entry_closed_long_ago_from_recently_closed_stucture(
+    struct aws_h2_connection *connection) {
+    uint64_t time_stamp;
+    if (aws_sys_clock_get_ticks(&time_stamp)) {
+        CONNECTION_LOG(ERROR, connection, "Failed getting the time stamp when stream closed");
+        return aws_h2err_from_last_error();
+    }
+    struct aws_array_list *closed_stream_array = &connection->thread_data.closed_streams_array;
+    uint32_t stream_id;
+    struct aws_hash_element *found = NULL;
+    /* iterate the array list from beginning, and remove all the entries closed long ago */
+    while (!aws_array_list_front(closed_stream_array, &stream_id)) {
+        const void *stream_id_key = (void *)(size_t)stream_id;
+        aws_hash_table_find(&connection->thread_data.closed_streams_map, stream_id_key, &found);
+        if (!found) {
+            CONNECTION_LOG(
+                ERROR,
+                connection,
+                "Internal error, stream id recorded in closed_stream array cannot be found in the hash map.");
+            return aws_h2err_from_h2_code(AWS_H2_ERR_INTERNAL_ERROR);
+        }
+        struct aws_h2_stream_closed_detail *closed_detail = found->value;
+        if (time_stamp > AWS_H2_IGNORE_TIME + closed_detail->closed_time_stamp) {
+            /* Probably remove the stream from dependency tree here, if we support PRIORITY */
+            /* Remove the element from the hash table and the array list, those options will not fail */
+            aws_hash_table_remove(&connection->thread_data.closed_streams_map, stream_id_key, NULL, NULL);
+            aws_array_list_pop_front(closed_stream_array);
+        } else {
+            break;
+        }
+    }
+    return AWS_H2ERR_SUCCESS;
+}
+
 static int s_record_closed_stream(
     struct aws_h2_connection *connection,
     uint32_t stream_id,
     enum aws_h2_stream_closed_when closed_when) {
 
     AWS_PRECONDITION(aws_channel_thread_is_callers_thread(connection->base.channel_slot->channel));
 
-    /* Peer might have already sent/queued frames for this stream before learning that we closed it.
-     * But if peer was the one to close the stream via RST_STREAM, they know better than to send more frames. */
-    bool frames_might_trickle_in = closed_when != AWS_H2_STREAM_CLOSED_WHEN_RST_STREAM_RECEIVED;
-    if (frames_might_trickle_in) {
-        if (aws_hash_table_put(
-                &connection->thread_data.closed_streams_where_frames_might_trickle_in,
-                (void *)(size_t)stream_id,
-                (void *)(size_t)closed_when,
-                NULL)) {
-
-            CONNECTION_LOG(ERROR, connection, "Failed inserting ID into map of recently closed streams");
-            return AWS_OP_ERR;
-        }
+    uint64_t time_stamp;
```

Comment:
timestamp is typically one word in the rest of the codebase, should probably be consistent with that

Reply:
Will do